### PR TITLE
ST: Improve Connect tests with Connector resource and check values set by…

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Enhancement / new feature

### Description

This PR extends original system test for Kafka Connector CR. 

Connector CR Status is tested as part of `testKafkaConnectS2IStatus` and `testKafkaConnectStatus` tests to save time for regression suite (to avoid duplicity test case which takes 6-8 minutes).

https://github.com/strimzi/strimzi-kafka-operator/issues/2354 is not tested as part of this PR. I suggest to create new with enhancement for current tests when it's fixed.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST,CustomResourceStatusST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=io.strimzi.systemtest.ConnectST#testKafkaConnectAndConnectorFileSinkPlugin,io.strimzi.systemtest.CustomResourceStatusST#testKafkaConnectS2IStatus test_only

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java [hunk 1]
```diff
@@ -170,40 +173,64 @@ void testKafkaBridgeStatus() {
...
+        KafkaConnectorResource.kafkaConnector(CLUSTER_NAME, TOPIC_NAME).done();
+        waitForKafkaConnectorStatus(CLUSTER_NAME, "Ready");
         assertKafkaConnectStatus(1, connectUrl);
 
         KafkaConnectResource.replaceKafkaConnectResource(CLUSTER_NAME, kb -> kb.getSpec().setResources(new ResourceRequirementsBuilder()
                 .addToRequests("cpu", new Quantity("100000000m"))
                 .build()));
         waitForKafkaConnectStatus("NotReady");
 
+        KafkaConnectorResource.replaceKafkaConnectorResource(CLUSTER_NAME,
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: I'm not sure I understand the logic why this is here. It doesn't seem to belong to this cluster and the actual Connect cluster is anyway not ready. I think the test with invalid lbel should be done when the Connect is ready.

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java [hunk 2]
```diff
@@ -170,40 +173,64 @@ void testKafkaBridgeStatus() {
...
         assertKafkaConnectS2IStatus(1, connectS2IUrl, connectS2IDeploymentConfigName);
+        KafkaConnectorResource.kafkaConnector(CONNECTS2I_CLUSTER_NAME, TOPIC_NAME).done();
+        waitForKafkaConnectorStatus(CONNECTS2I_CLUSTER_NAME, "Ready");
 
         KafkaConnectS2IResource.replaceConnectS2IResource(CONNECTS2I_CLUSTER_NAME, kb -> kb.getSpec().setResources(new ResourceRequirementsBuilder()
                 .addToRequests("cpu", new Quantity("100000000m"))
                 .build()));
         waitForKafkaConnectS2IStatus("NotReady");
 
+        KafkaConnectorResource.replaceKafkaConnectorResource(CONNECTS2I_CLUSTER_NAME,
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: Same as above.

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: Connector itself doesn't reflect status of Kafka Connect so I think it doesn't matter where this code is. I can move to test phase, where connect is ready with same results I think. Or I can create completely new test case, but I wanted to save some time during execution.

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: Why does it not reflect it? I do not think there is any guarantee that the connector can be `Ready` if the Connect is not `Ready`.

I understand why saving the time might be nice. But you should do this test when the Connect is `Ready`. That does not have to be different test case, or? You can just move it few lines up or down.

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: It's better question for @tombentley. After our last discussion I understood that Connector doesn't reflect connect status atm. Sure, I will change it.

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: It doesn't have to reflect Connect status. But there is a clear relationship. The Connector runs inside Connectc, so when Connect is not ready, where would you expect it to run?

Comment (tombentley) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: If the connector status is ready then it's very likely that the connect status is ready. If the connector status is not ready then you cannot infer anything about the connect status.

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: It is more the other direction: Does it make sense to test how new connector deployes when Connect is not ready?

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java (hunk 2 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: Moved to phase when Kafka Connect is ready

File: systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/CustomResourceStatusST.java]: Moved to phase when Kafka Connect is ready

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=CustomResourceStatusST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=CustomResourceStatusST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=CustomResourceStatusST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=io.strimzi.systemtest.CustomResourceStatusST#testKafkaConnectS2IStatus test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=CustomResourceStatusST,ConnectS2IST test_only

**Comment by Frawless** [comment]
@strimzi-ci run tests profile=regression testcase=io.strimzi.systemtest.ConnectS2IST#testDeployS2IWithMongoDBPlugin test_only

**Comment by Frawless** [comment]
This should be cherry-picked for the 0.16.0 release.

**Pull request merged**